### PR TITLE
Feature/distance gradient

### PR DESCRIPTION
**Pull request by tpoisot**
Do not merge - this is `DistanceGradient` but there is a problem we need to solve before.

### CODE REVIEW

**Comment by tpoisot** [comment]
Here we go!

index

File: src/algorithms/distancegradient.jl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+struct DistanceGradient <: NeutralLandscapeMaker
+    sources::Vector{Integer}
+end
+
+DistanceGradient() = DistanceGradient([1])
+
+function _landscape!(mat, alg::DistanceGradient)
+    @assert maximum(alg.sources) <= length(mat)
+    @assert minimum(alg.sources) > 0
+    indices = vcat(CartesianIndices(mat)...)
```
Comment (mkborregaard) [inline src/algorithms/distancegradient.jl]: `reduce(vcat, CartesianIndices(mat))`  is a more performant version - the splatting will be very slow for matrices of any relevant size.

**Review by mkborregaard — COMMENTED** [review]
I've got a number of thoughts here.
1. NearestNeighbours works for points in any location. I've always thought that for regular grids, it must be possible to define nearest neighbors (kdtrees) algorithmically.
2. I think the most common use case for this package is for simulation analyses - in which case you might often generate multiple random landscapes from the same object? Would it make sense for objects like this, that requires a lot of preprocessing, to have a type that catches e.g. the kdtree, making new random matrices fast?

**Comment by mkborregaard** [comment]
I am stupid. To get the results we need, the KDTree must be built on the sources of course. So in answers to above:
1. It does not look like there is such an algorithm (I've also asked around)
2. It makes no sense to cache the KDTree when it is built on the sources.

So that's that. I've made up for it by making a PR against this PR (#20) which is more performant and (hopefully) has a little cleaner code.

**Pull request merged**